[manual] Merge release/6.0-staging into release/6.0

carlossanlop opened the pull request:
## Merge commit!

Note: Draft because still waiting to include https://github.com/dotnet/runtime/pull/108300 https://github.com/dotnet/runtime/pull/107828 and https://github.com/dotnet/runtime/pull/108797

No OOB packages turned off in this release as we are turning them all on with https://github.com/dotnet/runtime/pull/108797 .

carlossanlop committed:
Merge pull request #107592 from dotnet/merge/release/6.0-to-release/6.0-staging

[automated] Merge branch 'release/6.0' => 'release/6.0-staging'

amanasifkhalid committed:
[release/6.0-staging] Revert `Linux_arm` images to `ubuntu-16.04` (#108396)

richlander committed:
Update RedHat 7 -> 8 -- release/6.0-staging (#105063)

* Update RedHat version

* Replace RHEL7 with CentOS Stream 9

* Update runtime queues

carlossanlop committed:
Merge branch 'release/6.0-staging' into merge/release/6.0-to-release/6.0-staging

carlossanlop committed:
Merge pull request #108496 from dotnet/merge/release/6.0-to-release/6.0-staging

[automated] Merge branch 'release/6.0' => 'release/6.0-staging'

carlossanlop committed:
Merge pull request #108714 from dotnet/merge/release/6.0-to-release/6.0-staging

[automated] Merge branch 'release/6.0' => 'release/6.0-staging'

dotnet-maestro[bot] committed:
[release/6.0-staging] Update dependencies from dotnet/xharness (#107756)

* Update dependencies from https://github.com/dotnet/xharness build 20240820.2

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 6.0.0-prerelease.24224.2 -> To Version 6.0.0-prerelease.24420.2

* Update dependencies from https://github.com/dotnet/xharness build 20240917.2

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 6.0.0-prerelease.24420.2 -> To Version 6.0.0-prerelease.24467.2

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Carlos Sánchez López <1175054+carlossanlop@users.noreply.github.com>

dotnet-maestro[bot] committed:
[release/6.0-staging] Update dependencies from dotnet/icu (#107764)

* Update dependencies from https://github.com/dotnet/icu build 20240912.2

Microsoft.NETCore.Runtime.ICU.Transport
 From Version 6.0.0-rtm.24315.1 -> To Version 6.0.0-rtm.24462.2

* Update dependencies from https://github.com/dotnet/icu build 20241010.4

Microsoft.NETCore.Runtime.ICU.Transport
 From Version 6.0.0-rtm.24462.2 -> To Version 6.0.0-rtm.24510.4

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Carlos Sánchez López <1175054+carlossanlop@users.noreply.github.com>

dotnet-maestro[bot] committed:
[release/6.0-staging] Update dependencies from dotnet/hotreload-utils (#107826)

* Update dependencies from https://github.com/dotnet/hotreload-utils build 20240914.1

Microsoft.DotNet.HotReload.Utils.Generator.BuildTool
 From Version 6.0.0-alpha.0.24402.2 -> To Version 6.0.0-alpha.0.24464.1

* Update dependencies from https://github.com/dotnet/hotreload-utils build 20241011.1

Microsoft.DotNet.HotReload.Utils.Generator.BuildTool
 From Version 6.0.0-alpha.0.24464.1 -> To Version 6.0.0-alpha.0.24511.1

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Carlos Sánchez López <1175054+carlossanlop@users.noreply.github.com>

ericstj reviewed (approved):
Requires https://github.com/dotnet/runtime/pull/108797

ericstj committed:
[release/6.0] Reduce net core app current package dependencies, increase direct update availability (#108797)

* Avoid package dependencies on libraries in the shared frameworkWe can avoid these dependencies since we can count on the library being part of the shared framework. Fewer dependencies means less packages downloaded, less for customers to service, less copied into the output directory when serviced.Backport of 6e440de5b435aee32673376773c58652d1474f49

* Enable packages for every project that reduced dependencies

* Add an option to enable servicing for transitive dependencies

* Permit settting ServiceTransitiveDependencies in projects as well

* Add additional packages required for up-stack servicing

* Fix indenting

Co-authored-by: Carlos Sánchez López <1175054+carlossanlop@users.noreply.github.com>

* Bump servicing version for packages that shipped last month

---------

Co-authored-by: Carlos Sánchez López <1175054+carlossanlop@users.noreply.github.com>

dotnet-maestro[bot] committed:
[release/6.0-staging] Update dependencies from dotnet/runtime-assets (#107828)

* Update dependencies from https://github.com/dotnet/runtime-assets build 20240913.1

Microsoft.DotNet.CilStrip.Sources , System.ComponentModel.TypeConverter.TestData , System.Data.Common.TestData , System.Drawing.Common.TestData , System.IO.Compression.TestData , System.IO.Packaging.TestData , System.Net.TestData , System.Private.Runtime.UnicodeData , System.Runtime.Numerics.TestData , System.Runtime.TimeZoneData , System.Security.Cryptography.X509Certificates.TestData , System.Windows.Extensions.TestData
 From Version 6.0.0-beta.24362.1 -> To Version 6.0.0-beta.24463.1

* Update dependencies from https://github.com/dotnet/runtime-assets build 20241010.5

Microsoft.DotNet.CilStrip.Sources , System.ComponentModel.TypeConverter.TestData , System.Data.Common.TestData , System.Drawing.Common.TestData , System.IO.Compression.TestData , System.IO.Packaging.TestData , System.Net.TestData , System.Private.Runtime.UnicodeData , System.Runtime.Numerics.TestData , System.Runtime.TimeZoneData , System.Security.Cryptography.X509Certificates.TestData , System.Windows.Extensions.TestData
 From Version 6.0.0-beta.24463.1 -> To Version 6.0.0-beta.24510.5

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Carlos Sánchez López <1175054+carlossanlop@users.noreply.github.com>

dotnet-maestro[bot] committed:
[release/6.0-staging] Update dependencies from dotnet/arcade (#108300)

* Update dependencies from https://github.com/dotnet/arcade build 20240913.1

Microsoft.DotNet.ApiCompat , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 6.0.0-beta.24413.1 -> To Version 6.0.0-beta.24463.1

* Update dependencies from https://github.com/dotnet/arcade build 20241008.2

Microsoft.DotNet.ApiCompat , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 6.0.0-beta.24463.1 -> To Version 6.0.0-beta.24508.2

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Carlos Sánchez López <1175054+carlossanlop@users.noreply.github.com>
